platform/src/main/java/org/hillview/sketches/DyadicHistogramBuckets.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2019 VMware Inc. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hillview.sketches;
+
+import org.hillview.table.api.IColumn;
+
+import java.util.Arrays;
+
+/**
+ * DyadicHistogramBuckets ensures that every bucket is a multiple of the leaf granularity.
+ * It is built on data that can be converted to doubles.
+ * The last bucket is right-inclusive.
+ */
+public class DyadicHistogramBuckets implements IHistogramBuckets {
```

Comment:
The comment will have to be much more elaborate; you can include a link to the paper with the binary mechanism.
You have to explain what the leaves are and what the rules are for creating the buckets.